lib/cadence/workflow/history.rb

```diff
@@ -15,6 +15,26 @@ def last_completed_decision_task
         events.select { |event| event.type == 'DecisionTaskCompleted' }.last
       end
 
+      def last_failed_decision_task
+        events.select { |event| event.type == 'DecisionTaskFailed' }.last
+      end
+
+      def last_timed_out_decision_task
+        events.select { |event| event.type == 'DecisionTaskTimedOut' }.last
+      end
+
+      def last_completed_activity_task
+        events.select { |event| event.type == 'ActivityTaskCompleted' }.last
+      end
+
+      def last_failed_activity_task
+        events.select { |event| event.type == 'ActivityTaskFailed' }.last
+      end
+
+      def last_timed_out_activity_task
```

Comment:
There's very little benefit of having these methods and it increases the public surface of the class, which is normally a bad sign. I think filtering events should be the responsibility of a developer using the History object unless there's something a bit more complex like grouping history by operations and their outcomes, similar to what "compact view" in Cadence UI does

Reply:
yeah I see that. I also realize that for one workflow, you can just look in the UI. I think I'll batch this for resetting a large amount of workflows though